### PR TITLE
[DOCS] Ingest processors docs improvements

### PR DESCRIPTION
**Pull request by leemthompo**
h/t @linyaru for the excellent proposal to improve the ingest processors overview.
Docs issue: https://github.com/elastic/search-docs-team/issues/32 

### Commits:

- [Categorize ingest processors on overview page, summarize use cases](https://github.com/elastic/elasticsearch/commit/85361ad60fcba2eeb2c0a80f5edfc7ab58f2b2fa) 
- [Add introductory info, subheading, links](https://github.com/elastic/elasticsearch/pull/104384/commits/faadbd0b8ab514feed2eda4a4118d2bca9bec5e1)

### ✨  [Screenshot](https://github.com/elastic/elasticsearch/assets/32779855/68fe5ce9-302f-4935-a6c9-91119373fd27)

([Live preview ](https://github.com/elastic/elasticsearch/pull/104384#issuecomment-1892604924)will be available once get a green CI)

### Food for thought 🤔 

Maybe this info would be more digestible in table format? 

[example](https://github.com/elastic/elasticsearch/assets/32779855/711da091-f2d6-4e60-b8f2-b7d99668e4cc)

### CODE REVIEW

**Review by szabosteve — APPROVED** [review]
Great content! Left a few minor suggestions. Please take or leave them. LGTM!

File: docs/reference/ingest/processors.asciidoc [hunk 1]
```diff
@@ -4,19 +4,207 @@
...
+<<convert-processor, `convert` processor>>::
+Converts a field in the currently ingested document to a different type, such as converting a string to an integer.
+
+<<dissect-processor, `dissect` processor>>::
+Extracts structured fields out of a single text field within a document.
+Unlike the <<grok-processor,grok processor>>, dissect does not use regular expressions.
+This makes the dissect's a simpler and often faster alternative.
+
+<<grok-processor, `grok` processor>>::
+Extracts fields via structured text parsing and pattern matching.
```
Comment (leemthompo) [inline docs/reference/ingest/processors.asciidoc:117]:
```suggestion
Extracts structured fields out of a single text field within a document using a regular expression dialect that supports reusable aliased expressions.
```

[the remark this answers is not in the document]
Comment (leemthompo) [reply docs/reference/ingest/processors.asciidoc:117]: Good idea, updated per [this comment](https://github.com/elastic/elasticsearch/pull/104384/files#r1453468290)

File: docs/reference/ingest/processors.asciidoc [hunk 2]
```diff
@@ -4,19 +4,207 @@
...
+
+<<set-processor, `set` processor>>::
+Sets a value on a field.
+
+[discrete]
+[[ingest-process-category-data-transformation-specific]]
+==== Specific outcomes
+
+<<bytes-processor, `bytes` processor>>::
+Converts a human readable byte value (e.g. 1kb) to its value in bytes (e.g. 1024).
```
[the remark this answers is not in the document]
Comment (leemthompo) [reply docs/reference/ingest/processors.asciidoc:136]: Makes sense, also reworked phrase to use just one parenthesis

File: docs/reference/ingest/processors.asciidoc [hunk 3]
```diff
@@ -4,19 +4,207 @@
...
+
+<<set-processor, `set` processor>>::
+Sets a value on a field.
+
+[discrete]
+[[ingest-process-category-data-transformation-specific]]
+==== Specific outcomes
+
+<<bytes-processor, `bytes` processor>>::
+Converts a human-readable byte value to its value in bytes (for example`1kb` becomes `1024`).
```
Comment (leemthompo) [inline docs/reference/ingest/processors.asciidoc:136]:
```suggestion
Converts a human-readable byte value to its value in bytes (for example `1kb` becomes `1024`).
```

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.12|[](https://github.com/elastic/elasticsearch/pull/104453)|
|✅|8.11|[](https://github.com/elastic/elasticsearch/pull/104454)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**